source/microbit/mprun.c

```diff
@@ -105,9 +118,11 @@ void mp_run(void) {
     readline_init0();
     microbit_init();
 
-    if (APPENDED_SCRIPT->header[0] == 'M' && APPENDED_SCRIPT->header[1] == 'P') {
+    if (hard_reboot && APPENDED_SCRIPT->header[0] == 'M' && APPENDED_SCRIPT->header[1] == 'P') {
```

Comment:
I think we do need this change. Otherwise how do you upload a file? 
If every time you ctrl-C the program restarts, there is no way to get into the repl.